handler/schedule_test.go

```diff
@@ -146,12 +146,161 @@ func TestScheduleHandler_mergeResourcesWithDefaults(t *testing.T) {
 			cfg.Config.GlobalMemoryResourceLimit = tt.globalMemoryResourceLimit
 			cfg.Config.GlobalMemoryResourceRequest = tt.globalMemoryResourceRequest
 			require.NoError(t, cfg.Config.ValidateSyntax())
-			s := ScheduleHandler{schedule: &v1alpha1.Schedule{Spec: v1alpha1.ScheduleSpec{
-				ResourceRequirementsTemplate: tt.template,
+			schedule := ScheduleHandler{schedule: &v1alpha1.Schedule{Spec: v1alpha1.ScheduleSpec{
+				ResourceRequirementsTemplate: tt.givenScheduleTemplate,
 			}}}
-			spec := &v1alpha1.RunnableSpec{Resources: tt.resources}
-			s.mergeResourcesWithDefaults(spec)
-			assert.Equal(t, tt.expected, spec.Resources)
+			res := &v1alpha1.RunnableSpec{
+				Resources: tt.givenResourceTemplate,
+			}
+			schedule.mergeResourcesWithDefaults(res)
+			assert.Equal(t, tt.expectedTemplate, res.Resources)
+		})
+	}
+}
+
+func TestScheduleHandler_mergeBackendWithDefaults(t *testing.T) {
+	tests := []struct {
```

Comment:
just a minor thing to be ready for our reference tests, would you mind changing this to a struct and move the `name` property of each struct to the key of the map?
```suggestion
	tests := map[string]struct {
```